torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1227,6 +1317,8 @@ void ProcessGroupNCCL::workCleanupLoop() {
         lock,
         std::chrono::milliseconds(kWatchdogThreadSleepMillis),
         [&]() -> bool { return terminateProcessGroup_.load(); });
+    // Bump up heart beat by one.
+    heartbeat_++;
 
     for (auto it = workMetaList_.begin(); it != workMetaList_.end();
```

Comment:
We probably need to handle the 'desync report' case.

if we go into desync report mode below, desync report may take long enough that heartbeat monitor will kill us.

Reply:
OK, so I am going to add a another flag for desync report.